internal/kubernetes/deploy_test.go

```diff
@@ -0,0 +1,37 @@
+package kubernetes
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/numaproj-labs/numaplane/tests/utils"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestToUnstructured(t *testing.T) {
```

Comment:
Can you add test cases for Apply/Delete resources as well? Is it possible to do it with the fake k8s client?

Reply:
This can be a follow up though.

Reply:
yes, I'll have a follow up PR for fake k8s client as I have added TODO here https://github.com/numaproj-labs/numaplane/pull/44/files#diff-321de43a0cf1c625281d5f154b8fb166a5f4984578474d9bf0758dd4a2aab75dR18